applications/sintering/include/pf-applications/sintering/operator_sintering_generic.h

```diff
@@ -270,13 +270,31 @@ namespace Sintering
 
     template <unsigned int with_time_derivative = 2>
     void
-    evaluate_nonlinear_residual(BlockVectorType &      dst,
-                                const BlockVectorType &src) const
+    evaluate_nonlinear_residual(
+      BlockVectorType &      dst,
+      const BlockVectorType &src,
+      const std::function<void(const unsigned int, const unsigned int)>
+        pre_operation = {},
+      const std::function<void(const unsigned int, const unsigned int)>
+        post_operation = {}) const
```

Comment:
What is the final gain of these pre/post optimizations?

Reply:
The goal is to overlap the cell loop with vector updates as done in https://journals.sagepub.com/doi/abs/10.1177/10943420221107880.